pkg/network/usm/ebpf_ssl_test.go

```diff
@@ -73,17 +74,46 @@ func TestNativeShortLivedProcess(t *testing.T) {
 	libmmap := filepath.Join(curDir, "testdata", "site-packages", "ddtrace")
 	lib := filepath.Join(libmmap, fmt.Sprintf("libssl.so.%s", runtime.GOARCH))
 
+	var pidMap sync.Map
+	utils.RegisterHook = func(pid uint32) {
+		pidMap.Store(pid, struct{}{})
+	}
+	t.Cleanup(func() {
+		// Will be called after monitor is stopped (due to t.Cleanup() ordering)
+		// so should be race-free.
+		utils.RegisterHook = nil
+	})
```

Comment:
We don't actually "care" of nativeTLS here, we want to check the file_registry handles the scenario correctly, so I do think we can create a test in `file_registry_test.go` can provides custom activation and deactivation callbacks

It will provide a cleaner test, less chances for flakiness, faster test